core/shared/src/main/scala/zio/prelude/data/Optional.scala

```diff
@@ -119,11 +119,8 @@ sealed trait Optional[+A] { self =>
       case Optional.Absent       => Optional.Absent
     }
 
-  final def orElse[B >: A](other: Optional[B]): Optional[B] =
-    self match {
-      case Optional.Present(_) => self
-      case Optional.Absent     => other
-    }
+  final def orElse[B >: A](alternative: => Optional[B]): Optional[B] =
+    if (isEmpty) alternative else this
```

Comment:
Code copied from `Option#orElse`. Avoids the call to the `unapply` function in `Optional.Present(_)`